tests/purs/passing/FFIConstraintWorkaround.purs

```diff
@@ -0,0 +1,19 @@
+module FFIConstraintWorkaround where
```

Comment:
This needs to be `module Main`, and it's a little alarming that our test fixture doesn't catch that (and, I think, just silently re-runs the previously compiled test).

Anyway, after you fix that, I think you'll find that this test doesn't pass because the output doesn't end in ‘Done’. Might be better to explicitly compare the result of `showFFI` to the expected strings, since in the counterfactual regression I was worried about where `show` is passed before being applied to a dictionary, what I think would happen with this test as written is that the value would be passed as the first JavaScript argument to `show`, which means it would be considered the dictionary, and `show` would silently accept that up until it receives a second argument which never arrives. So `showFFI` would return a JavaScript function that PureScript thinks is a `String` and happily passes to `log`; `log` probably prints `[Function (anonymous)]` to stdout, and the test continues on its way none the wiser.

Reply:
I've addressed this comment in latest commits.

Reply:
#4243 should fix the `Main` module issue.